.github/copilot-instructions.md

```diff
@@ -0,0 +1,238 @@
+# IDE Sidecar - AI Coding Agent Instructions
+
+## Project Overview
+
+IDE Sidecar is a **Java 21 Quarkus application** that serves as the backend for Confluent's VS Code extension. It exposes REST and GraphQL APIs for managing connections to Kafka clusters and consuming/producing messages. The project is compiled to a **native executable using GraalVM** for fast startup and low memory footprint.
+
+**Key Connection Types:**
+- `CCLOUD`: Confluent Cloud connections (OAuth-based authentication)
+- `LOCAL`: Confluent Local/standalone Kafka
+- `DIRECT`: Direct connections to Kafka clusters (supports multiple auth methods: Basic, SASL/SCRAM, Kerberos, OAuth, TLS)
+
+## Architecture
+
+### Processor Chain Pattern
```

Comment:
I guess this section is referring to [reactive routes](https://quarkus.io/guides/reactive-routes). It's mainly used in the context of the proxies. This should not be the first section under Architecture, from my POV.

Reply:
Makes sense, I will move under "Reactive Programming"